controllers/extension-certificate-service/pkg/controller/certservice/actuator.go

```diff
@@ -0,0 +1,306 @@
+	}
+	a.applier = applier
+	return nil
+}
+
+// InjectClient injects the controller runtime client into the reconciler.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+var certBrokerChartPath = filepath.Join(utils.ChartsPath, certBrokerName)
+
+const certBrokerName = "cert-broker"
+
+func (a *actuator) createCertBroker(ctx context.Context, shoot *gardenv1beta1.Shoot, namespace string) error {
+	shootDomain := shoot.Spec.DNS.Domain
+	if shootDomain == nil {
+		return fmt.Errorf("no domain given for shoot %s/%s", shoot.GetName(), shoot.GetNamespace())
+	}
+
+	var (
+		dns        []map[string]string
+		configSpec = a.certServiceConfig.Spec
+	)
+
+	for _, route53Provider := range configSpec.Providers.Route53 {
+		route53values := internal.CreateDNSProviderValue(&route53Provider, *shootDomain)
+		if route53values != nil {
+			dns = append(dns, route53values)
+		}
+	}
+	for _, cloudDNSProvider := range configSpec.Providers.CloudDNS {
+		cloudDNSValues := internal.CreateDNSProviderValue(&cloudDNSProvider, *shootDomain)
+		dns = append(dns, cloudDNSValues)
+	}
+
+	shootKubeconfig, err := a.createKubeconfigForCertManager(ctx, namespace)
+	if err != nil {
+		return err
+	}
+
+	shootKubeconfigChecksum := util.ComputeChecksum(shootKubeconfig.Data)
+
+	certBrokerConfig := map[string]interface{}{
+		"replicas": util.GetReplicaCount(shoot, 1),
+		"certbroker": map[string]interface{}{
+			"targetClusterSecret": shootKubeconfig.GetName(),
+		},
+		"certmanager": map[string]interface{}{
+			"clusterissuer": configSpec.IssuerName,
+			"dns":           dns,
+		},
+		"podAnnotations": map[string]interface{}{
+			"checksum/secret-cert-broker": shootKubeconfigChecksum,
+		},
+	}
+
+	certBrokerConfig, err = imagevector.ImageVector().InjectImages(certBrokerConfig, "", "", certBrokerName)
+	if err != nil {
+		return fmt.Errorf("failed to find image version for %s: %v", certBrokerName, err)
+	}
+
+	a.logger.Info("Component is being applied", "component", "cert-broker", "namespace", namespace)
+	return a.applier.ApplyChartInNamespace(
+		ctx,
+		certBrokerChartPath,
+		namespace,
+		certBrokerName,
+		certBrokerConfig,
+		nil,
+	)
+}
+
+func (a *actuator) deleteCertBroker(ctx context.Context, namespace string) error {
+	meta := metav1.ObjectMeta{
+		Name:      utils.CertBrokerResourceName,
+		Namespace: namespace,
+	}
+
+	objects := []runtime.Object{
+		&appsv1.Deployment{
+			ObjectMeta: meta,
+		},
+		&corev1.Secret{
+			ObjectMeta: meta,
+		},
+		&corev1.ServiceAccount{
+			ObjectMeta: meta,
+		},
+		&rbacv1.RoleBinding{
+			ObjectMeta: meta,
+		},
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: meta,
+		},
+	}
+
+	a.logger.Info("Component is being deleted", "component", "cert-broker", "namespace", namespace)
+	for _, obj := range objects {
+		if err := a.client.Delete(ctx, obj); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	secret, err := a.createKubeconfigForCertManager(ctx, namespace)
```

Comment:
Why not exporting the name of this kubeconfig as a constant somewhere and simply add another `&corev1.Secret` object to your list of objects above?